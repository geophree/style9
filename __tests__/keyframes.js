/* eslint-env jest */
const compile = require('./compile.js');

it('supports keyframes', () => {
  const input = `
import style9 from 'style9';
style9.keyframes({
  '0%': {
    color: 'blue'
  },
  '100%': {
  }
});
  `;
  const { code, styles } = compile(input);

  expect(code).toMatchSnapshot();
  expect(styles).toMatchSnapshot();
});

it('converts from', () => {
  const input = `
import style9 from 'style9';
style9.keyframes({
  from: {
    color: 'blue'
  }
});
  `;
  const { code, styles } = compile(input);

  expect(code).toMatchSnapshot();
  expect(styles).toMatchSnapshot();
});

it('converts to', () => {
  const input = `
import style9 from 'style9';
style9.keyframes({
  to: {
    color: 'blue'
  }
});
  `;
  const { code, styles } = compile(input);

  expect(code).toMatchSnapshot();
  expect(styles).toMatchSnapshot();
});

it('supports complex styles and times', () => {
  const input = `
import style9 from 'style9';
style9.keyframes({
  '0%': { top: 0, left: 0 },
  '30%': { top: '50px', marginBottom: 1, margin: 0 },
  '68%, 72%': { left: '50px', marginBottom: 5 },
  '100%': { top: '100px', left: '100%' }
});
  `;
  const { code, styles } = compile(input);

  expect(code).toMatchSnapshot();
  expect(styles).toMatchSnapshot();
});

it('supports settings animationName directly', () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  default: {
    animationName: style9.keyframes({
      '0%': {
        opacity: 0
      }
    })
  }
});
styles.default
  `;
  const { code, styles } = compile(input);

  expect(code).toMatchSnapshot();
  expect(styles).toMatchSnapshot();
});
