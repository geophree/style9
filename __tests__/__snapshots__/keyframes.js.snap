// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts from 1`] = `
"import style9 from 'style9';
\\"clbvhnp\\";"
`;

exports[`converts from 2`] = `"@keyframes clbvhnp{0%{color:blue}}"`;

exports[`converts to 1`] = `
"import style9 from 'style9';
\\"c1yehff3\\";"
`;

exports[`converts to 2`] = `"@keyframes c1yehff3{100%{color:blue}}"`;

exports[`supports complex styles and times 1`] = `
"import style9 from 'style9';
\\"c1suwj5z\\";"
`;

exports[`supports complex styles and times 2`] = `"@keyframes c1suwj5z{0%{top:0px;left:0px}30%{top:50px;margin-bottom:1px;margin-top:0px;margin-right:0px;margin-left:0px}68%, 72%{left:50px;margin-bottom:5px}100%{top:100px;left:100%}}"`;

exports[`supports keyframes 1`] = `
"import style9 from 'style9';
\\"c1l2fbnf\\";"
`;

exports[`supports keyframes 2`] = `"@keyframes c1l2fbnf{0%{color:blue}100%{}}"`;

exports[`supports settings animationName directly 1`] = `
"import style9 from 'style9';
const styles = {
  default: {
    animationName: \\"csqo9ij\\"
  }
};
styles.default;"
`;

exports[`supports settings animationName directly 2`] = `"@keyframes c1px3r7c{0%{opacity:0}}.csqo9ij{animation-name:c1px3r7c}"`;
